sympy/core/relational.py

```diff
@@ -311,6 +311,9 @@ def _eval_simplify(self, **kwargs):
             v = None
             if dif.is_comparable:
                 v = dif.n(2)
+            elif hasattr(dif, 'is_empty'):
+                if dif.is_empty:
+                    v = S.Zero
```

Comment:
I'm not sure I understand what this does. Can you demonstrate?

Reply:
Earlier in case of sets, after `dif = lhs - rhs`  it tried to call `set.equal(0)`, i.e it tried to equate(considering `dif` = 0) empty set with 0 which was throwing error as `Sets` didn't have `.equal()` function , if that function is added, still comparing set with 0 would give error.

If `dif` is a set, empty or not it, would have `is_empty` attribute . Thus checking if `dif` is an empty set or not and proceeding and setting `v = S.Zero` accordingly.

In short in case of sets checking if `lhs - rhs`  is an `EmptySet` or not.

Reply:
We should avoid computing `dif` altogether unless both sides are instances of Expr.

Reply:
But `dif` is playing nice with sets and is working as expected. Are you sure changing the entire method for sets is a good idea?

Reply:
The logic in `Relational._eval_simplify` is clearly only intended for expressions. It is better to guard against any non-Expr inputs before calculating `dif`. Ideally we should restrict this block to number-type expressions but there isn't a clear way to do that.

For sets anyway it would probably be better to check equality using the subset relationship. Checking `(A - B)` is also not the same as checking `(B - A)` and it's possible that neither `A-B` or `B-A` is empty even though the sets are clearly unequal.

Reply:
Should I introduce handlers for specific cases like done for `is_subset` in sets?

Reply:
I don't think that we need multiple dispatch in `Relational._eval_simplify` because both sides should always be of the same "type" in some sense. The only possible simplification here is that we can return True or False if we know that the two sides are or are not equal and that we can just determine using `is_subset(A, B)` and `is_subset(B, A)`.